Automattic/wp-calypso
# Replace getRewindCapabilities with feature checks

#### Changes proposed in this Pull Request

* Updates remaining `getRewindCapabilities` calls with feature checks.
* Removes `getRewindCapabilities` selector.
* Removes unused `getAllowRestore` selector.

#### Testing instructions



* Navigate to https://container-frosty-leakey.calypso.live/backup/ and select a site
* With a site that has Realtime Backups (eCommerce) make sure the detail view is being shown: <img width="1510" alt="image" src="https://user-images.githubusercontent.com/1398304/170094976-3386cd4c-2718-43ec-b106-f5532240dea7.png">
* With a site on a lower tier plan, check that upgrade buttons are still shown.
* [Create a new jurassic.ninja site](https://jurassic.ninja/specialops/) with a backup subscription.
* Once connected, try purchasing Jetpack Security and make sure the warning banner shows: <img width="1510" alt="Screen Shot 2022-05-24 at 10 14 42 AM" src="https://user-images.githubusercontent.com/1398304/170095280-f69893d7-aeb1-47e3-8774-716cc65804ce.png">




Related to p4TIVU-a66-p2
